### PR TITLE
remove io related cruft from charts

### LINKED ISSUE #3132: restore `Chart(server=True)`
server=True mode was commented out at some point on master, it needs to be restored (or replaced with something that makes more sense with the new server, possibly? I haven't investigated the big picture).

See https://github.com/bokeh/bokeh/blob/2a3ad528279e3b329857d712b61fcca536a59cf8/bokeh/charts/chart.py#L123-L128

For how to fix it up: https://github.com/bokeh/bokeh/wiki/Porting-guide:-new-Bokeh-server

### PR DESCRIPTION
**Pull request by fpliger**
issues: fixes #3132

### CODE REVIEW

File: bokeh/charts/chart.py [hunk 1]
```diff
@@ -117,19 +118,6 @@ class Chart(Plot):
...
-    A name for the file to save this chart to.
-    """)
-
-    server = Either(Bool(False), String, help="""
-    A name to use to save this chart to on server.
-    """)
-
-    notebook = Either(Bool(False), String, help="""
-    Whether to display the plot inline in an IPython/Jupyter
-    notebook.
```
Comment (bryevdv) [inline bokeh/charts/chart.py]: we do have the notion of a deprecated property now, we might want to use that here to warn people. @canavandl can you point @fpliger at the deprecated property code/example?

Comment (fpliger) [inline bokeh/charts/chart.py]: Right. Filtering `kwargs` seems simpler, yes..

**Comment by fpliger** [comment]
the deprecation using our deprecated property was nicier (and better) then filtering `kwarts` so ended doing this way. Thanks for the suggestion :)

File: bokeh/charts/chart.py [hunk 2]
```diff
@@ -322,46 +298,71 @@ def make_grid(self, dimension, ticker):
...
-
-        if self.notebook:
-            from bokeh.embed import notebook_div
-            publish_display_data({'text/html': notebook_div(self)})
+    @property
+    def filename(self):
+        warnings.warn(
+            """
+            Chart property 'filename' will be deprecated in Bokeh
+            0.12.0. Use 'bokeh.io.output_file' instead.
```
Comment (bryevdv) [inline bokeh/charts/chart.py]: Better message is to say it "was deprecated in `0.11` and will be removed in the future". It's not certain there will be a `0.12`

**Pull request merged**